http-body-util/src/combinators/buffered.rs

```diff
@@ -0,0 +1,38 @@
+use std::{
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use futures_util::Future;
+use http_body::Body;
+use pin_project_lite::pin_project;
+
+pin_project! {
+    /// Future that resolves into a `Buffered`.
+    pub struct Buffered<T: ?Sized> {
+        #[pin]
+        pub(crate) body: T
+    }
+}
+
+impl<T: Body + Unpin + ?Sized> Future for Buffered<T> {
```

Comment:
Oh, this is interesting. I do think this way is easier in some sense to get the full frame set... I had imagined something with `async fn data()` that you'd await over and over, and then you could await `trailers`. I suppose what I just described is also possible here.

And people can use a `Buffered<Limited<Blah>>` to cap how much they're willing to read.

Cool!

Reply:
Yeah, I think maybe what we want this to be is what you're describing and then we provide a `Collected` type which is what this impl is?